storage/innobase/log/log0recv.cc

```diff
@@ -1640,11 +1639,11 @@ static byte *recv_parse_or_apply_log_rec_body(mlog_id_t type, byte *ptr,
                 fil_tablespace_redo_encryption(ptr, end_ptr, space_id, apply));
           } else if (memcmp(ptr_copy, ENCRYPTION_KEY_MAGIC_PS_V1,
                             ENCRYPTION_MAGIC_SIZE) == 0 &&
-                     apply) {
+                     !apply) {
             return (fil_parse_write_crypt_data_v1(space_id, ptr, end_ptr, len));
           } else if (memcmp(ptr_copy, ENCRYPTION_KEY_MAGIC_PS_V2,
                             ENCRYPTION_MAGIC_SIZE) == 0 &&
-                     apply) {
+                     !apply) {
             return (fil_parse_write_crypt_data_v2(space_id, ptr, end_ptr, len));
```

Comment:
For redo parsing tools like PXB, which create file, based on the redo records, the MLOG_FILE_CREATE should the first record. So please change the order of redo creation for crypt_data and MLOG_FILE_CREATE*

Reply:
Space create during parsing MLOG_FILE_CREATE will be opened with open_for_recovery - where we assign harversted (during parsing) crypt_data.